packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx

```diff
@@ -48,21 +54,68 @@ export interface DashboardLayoutSlots {
    * @default Link
    */
   appTitle?: React.ElementType;
+
   /**
    * The toolbar actions component used in the layout header.
    * @default ToolbarActions
+   * @see [DashboardLayout#slots](https://mui.com/toolpad/core/react-dashboard-layout/#slots)
    */
-  toolbarActions?: React.JSXElementConstructor<{}>;
+  appBarActions?: React.JSXElementConstructor<{}>;
+
   /**
-   * The toolbar account component used in the layout header.
+   * The toolbar actions component used in the layout header.
    * @default Account
+   * @deprecated it is moved to the toolbarActions component
+   * @see [DashboardLayout#slots](https://mui.com/toolpad/core/react-dashboard-layout/#slots)
    */
   toolbarAccount?: React.JSXElementConstructor<AccountProps>;
+
   /**
    * Optional footer component used in the layout sidebar.
    * @default null
    */
   sidebarFooter?: React.JSXElementConstructor<SidebarFooterProps>;
+
+  /**
+   * This component will completely override the internal layout of the top bar.
+   * You can still use the built-in `<AppTitle />`, `<ThemeSwitcher />` and `<Account />` components inside.
+   * @default null
+   * @see [DashboardLayout#slots](https://mui.com/toolpad/core/react-dashboard-layout/#slots)
+   * @example
+   * ```tsx
+   * function CustomAppBar(props: { menuIcon: React.ReactElement }) {
```

Comment:
This would ideally be made into its own importable component to spare the user from having to copy paste so many lines, and the example from taking so much space.

Reply:
Again all of these should be moved to something like a `DashboardLayoutHeader` internal component in its own file to be used and referenced here.

Same pattern we're following for the `ToolbarActions` component, for example.